File: webui-src/src/api/antiDetect.ts
import { client } from './client';

export interface AntiDetectConfig {
  enable_fingerprint: boolean;
  enable_rate_limit: boolean;
  enable_human_behavior: boolean;
  enable_account_health: boolean;
  enable_binding: boolean;
  fingerprint_platform_hint?: string;
  rate_limit_min_interval: number;
  rate_limit_max_interval: number;
  rate_limit_hourly_limit: number;
  rate_limit_daily_limit: number;
  account_cooling_threshold: number;
  account_warning_threshold: number;
}

export interface AntiDetectStats {
  platform: string;
  total_accounts: number;
  active_accounts: number;
  cooling_accounts: number;
  warning_accounts: number;
  banned_accounts: number;
  avg_risk_score: number;
  total_bindings: number;
  with_proxy: number;
  with_fingerprint: number;
  complete_bindings: number;
  total_requests: number;
  hourly_requests: number;
  daily_requests: number;
  hourly_limit: number;
  daily_limit: number;
}

export interface AccountHealthStatus {
  account_id: string;
  platform: string;
  total_requests: number;
  failed_requests: number;
  captcha_count: number;
  risk_score: number;
  status: 'active' | 'cooling' | 'warning' | 'banned';
  last_request_at?: string;
  cooling_until?: string;
  bound_proxy_id?: string;
  bound_fingerprint_id?: string;
}

export interface BindingInfo {
  account_id: string;
  platform: string;
  proxy_id?: string;
  fingerprint_id?: string;
  created_at: string;
  updated_at: string;
}

export const antiDetectApi = {
  // 获取统计信息
  getStats: async (platform: string): Promise<AntiDetectStats> => {
    const response = await client.get(`/api/anti-detect/stats/${platform}`);
    return response.data;
  },

  // 获取账号健康列表
  getAccountHealth: async (
    platform: string,
    page = 1,
    pageSize = 20
  ): Promise<{ accounts: AccountHealthStatus[]; total: number; page: number; page_size: number }> => {
    const response = await client.get(`/api/anti-detect/accounts/${platform}`, {
      params: { page, page_size: pageSize },
    });
    return response.data;
  },

  // 获取绑定列表
  getBindings: async (
    platform: string,
    page = 1,
    pageSize = 20
  ): Promise<{ bindings: BindingInfo[]; total: number; page: number; page_size: number }> => {
    const response = await client.get(`/api/anti-detect/bindings/${platform}`, {
      params: { page, page_size: pageSize },
    });
    return response.data;
  },

  // 绑定代理
  bindProxy: async (platform: string, accountId: string, proxyId: string): Promise<void> => {
    await client.post(`/api/anti-detect/bindings/${platform}/${accountId}/proxy`, null, {
      params: { proxy_id: proxyId },
    });
  },

  // 删除绑定
  deleteBinding: async (platform: string, accountId: string): Promise<void> => {
    await client.delete(`/api/anti-detect/bindings/${platform}/${accountId}`);
  },

  // 获取默认配置
  getDefaultConfig: async (platform?: string): Promise<AntiDetectConfig> => {
    const response = await client.get('/api/anti-detect/config/default', {
      params: platform ? { platform } : {},
    });
    return response.data;
  },
};

File: api/schemas/anti_detect.py
"""
反爬增强相关数据模型
"""
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field
from datetime import datetime


class AntiDetectConfig(BaseModel):
    """反爬增强配置"""
    # 功能开关
    enable_fingerprint: bool = True              # 启用浏览器指纹
    enable_rate_limit: bool = True               # 启用智能限速
    enable_human_behavior: bool = True           # 启用人类行为模拟
    enable_account_health: bool = True           # 启用账号健康管理
    enable_binding: bool = True                  # 启用账号-代理-指纹绑定

    # 指纹配置
    fingerprint_platform_hint: Optional[str] = None  # 平台提示: mac/windows/linux

    # 限速配置
    rate_limit_min_interval: float = 3.0         # 最小请求间隔（秒）
    rate_limit_max_interval: float = 10.0        # 最大请求间隔（秒）
    rate_limit_hourly_limit: int = 150           # 每小时最大请求数
    rate_limit_daily_limit: int = 1500           # 每天最大请求数

    # 账号健康配置
    account_cooling_threshold: float = 70.0      # 冷却阈值（风险评分）
    account_warning_threshold: float = 50.0      # 警告阈值（风险评分）


class AccountHealthStatus(BaseModel):
    """账号健康状态"""
    account_id: str
    platform: str
    total_requests: int = 0
    failed_requests: int = 0
    captcha_count: int = 0
    risk_score: float = 0.0
    status: str = "active"  # active/cooling/warning/banned
    last_request_at: Optional[datetime] = None
    cooling_until: Optional[datetime] = None
    bound_proxy_id: Optional[str] = None
    bound_fingerprint_id: Optional[str] = None


class BindingInfo(BaseModel):
    """绑定信息"""
    account_id: str
    platform: str
    proxy_id: Optional[str] = None
    fingerprint_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class AntiDetectStats(BaseModel):
    """反爬增强统计"""
    platform: str

    # 账号健康统计
    total_accounts: int = 0
    active_accounts: int = 0
    cooling_accounts: int = 0
    warning_accounts: int = 0
    banned_accounts: int = 0
    avg_risk_score: float = 0.0

    # 绑定统计
    total_bindings: int = 0
    with_proxy: int = 0
    with_fingerprint: int = 0
    complete_bindings: int = 0

    # 限速统计
    total_requests: int = 0
    hourly_requests: int = 0
    daily_requests: int = 0
    hourly_limit: int = 0
    daily_limit: int = 0


class AccountHealthListResponse(BaseModel):
    """账号健康列表响应"""
    accounts: List[AccountHealthStatus]
    total: int
    page: int
    page_size: int


class BindingListResponse(BaseModel):
    """绑定列表响应"""
    bindings: List[BindingInfo]
    total: int
    page: int
    page_size: int
